### ISSUE TITLE
Segmentation fault  (core dumped) 

### ISSUE THREAD

**Issue by moldach** [BGI-flexlab/SOAPnuke#53]
I'm getting an error trying to run SOAPnuke on raw/cleaned samples. Each FASTQ is ~101Gb in size (300,000,000 reads for each read pair)

```
/bin/bash
#BSUB -q normal
#BSUB -J SOAPnuke_BIG
#BSUB -n 1
##BSUB -R "span[hosts=1]"
#BSUB -R "select[hname!=node036]"
#BSUB -M 2500000
#BSUB -W 72:00
#BSUB -u moldach@ucalgary.ca
#BSUB -B
#BSUB -N
#BSUB -o SOAPnuke_script_BIG.out
#BSUB -e SOAPnuke_script_BIG.err

export LD_LIBRARY_PATH=/home/moldach/bin/SOAPnuke/

/home/moldach/bin/SOAPnuke/SOAPnuke filter \
        -1 ./data/Raw_Fastq/R1/S1aR1.fastq \
        -2 ./data/Raw_Fastq/R2/S1aR2.fastq \
        -C ./data/Trimmed_Fastq/R1/S1aR1.fastq \
        -D ./data/Trimmed_Fastq/R2/S1aR2.fastq \
        -o ./
```

LSF's e-mail is:

```
Exited with exit code 139.

Resource usage summary:

    CPU time :                                   7576.88 sec.
    Max Memory :                                 325 MB
    Average Memory :                             273.10 MB
    Total Requested Memory :                     -
    Delta Memory :                               -
    Max Swap :                                   2 MB
    Max Processes :                              8
    Max Threads :                                18
    Run time :                                   1498 sec.
    Turnaround time :                            1500 sec.
```

And the `.err` log is:

```
/home/moldach/.lsbatch/1595546086.86175.shell: line 22: 139963 Segmentation fault      (core dumped) /home/moldach/bin/SOAPnuke/SOAPnuke filter -1 ./data/Raw_Fastq/R1/S1aR1.fastq -2 ./data/Raw_Fastq/R2/S1aR2.fastq -C ./data/Trimmed_Fastq/R1/S1aR1.fastq -D ./data/Trimmed_Fastq/R2/S1aR2.fastq -o ./`
```

**Comment by berry08** [comment]
Which version of SOAPnuke do you use? We suggest keep updating the software, maybe it's a bug in a history version

**Comment by moldach** [comment]
Hi @berry08 I'm using Version 2.1.2 of SOAPnuke. I downloaded it this week via:

```
git clone https://github.com/BGI-flexlab/SOAPnuke.git
cd SOAPnuke
make
```

**Comment by berry08** [comment]
@moldach Could you send the core file to us to debug?

**Comment by moldach** [comment]
Here's a link to the core file: https://www.dropbox.com/s/wffit7zn3fmles9/core.21899?dl=0

**Comment by berry08** [comment]
> Here's a link to the core file: https://www.dropbox.com/s/wffit7zn3fmles9/core.21899?dl=0

Thank you. We have fixed this bug in 2.1.4.beta.1, please try again after updating software
